src/modm/platform/spi/sam/spi_master.cpp.in

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright (c) 2009, Martin Rosekeit
+ * Copyright (c) 2009-2012, Fabian Greif
+ * Copyright (c) 2010, Georgi Grinshpun
+ * Copyright (c) 2012-2017, Niklas Hauser
+ * Copyright (c) 2013, Kevin Läufer
+ * Copyright (c) 2014, Sascha Schade
+ * Copyright (c) 2021, Jeff McBride
+ *
+ * This file is part of the modm project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+// ----------------------------------------------------------------------------
+
+#include "spi_master_{{id}}.hpp"
+
+// Bit0: single transfer state
+// Bit1: block transfer state
+uint8_t
+modm::platform::SpiMaster{{ id }}::state(0);
```

Comment:
you can use `static inline uint8_t state{0};` etc in the class declaration since C++17. (We need to update our style.)